app/javascript/pages/Settings/Password/Show.tsx

```diff
@@ -0,0 +1,98 @@
+import { useForm, usePage } from "@inertiajs/react";
+import * as React from "react";
+import { cast } from "ts-safe-cast";
+
+import { SettingPage } from "$app/parsers/settings";
+
+import { Button } from "$app/components/Button";
+import { PasswordInput } from "$app/components/PasswordInput";
+import { showAlert } from "$app/components/server-components/Alert";
+import { Layout as SettingsLayout } from "$app/components/Settings/Layout";
+
+const MIN_PASSWORD_LENGTH = 4;
+const MAX_PASSWORD_LENGTH = 128;
+
+type PasswordPageProps = {
+  settings_pages: SettingPage[];
+  require_old_password: boolean;
+};
+
+export default function PasswordPage() {
+  const props = cast<PasswordPageProps>(usePage().props);
+  const uid = React.useId();
+  const [requireOldPassword, setRequireOldPassword] = React.useState(props.require_old_password);
+
+  const form = useForm({
+    user: {
+      password: "",
+      new_password: "",
+    },
+  });
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (form.data.user.new_password.length < MIN_PASSWORD_LENGTH) {
+      showAlert("Your new password is too short.", "error");
+      return;
+    }
+
+    if (form.data.user.new_password.length >= MAX_PASSWORD_LENGTH) {
+      showAlert("Your new password is too long.", "error");
+      return;
+    }
+
+    form.put(Routes.settings_password_path(), {
+      preserveScroll: true,
+      onSuccess: (response) => {
+        if (response.props.new_password) setRequireOldPassword(true);
+        form.reset();
+      },
+    });
+  };
```

Comment:
again the file uses exact same code as the component which was used previously, just shifted the form submission flow out of inline to a function and also using inertia forms for it